client/lib/source/poll-source.ts

```diff
@@ -23,33 +14,13 @@ export class PollSource<T> implements Source<T> {
     private readonly http: Http<T>,
   ) {}
 
-  watch$(store: Store<T>): Observable<RPCResponse<Update<T>>> {
-    const polling$ = new BehaviorSubject('')
-
-    const updates$ = of({}).pipe(
-      concatMap(_ => store.sequence$),
+  watch$({ sequence$ }: Store<T>): Observable<Update<T>> {
+    return sequence$.pipe(
       concatMap(seq => this.http.getRevisions(seq)),
       take(1),
-    )
-
-    const delay$ = of([]).pipe(
-      delay(this.pollConfig.cooldown),
-      tap(_ => polling$.next('')),
-      skip(1),
-    )
-
-    const poll$ = concat(updates$, delay$)
-
-    return polling$.pipe(
-      switchMap(_ => poll$),
-      concatMap(res => {
-        if (Array.isArray(res)) {
-          return from(res) // takes Revision[] and converts it into Observable<Revision>
-        } else {
-          return of(res) // takes Dump<T> and converts it into Observable<Dump<T>>
-        }
-      }),
-      map(result => ({ result, jsonrpc: '2.0' })),
+      // convert Revision[] it into Observable<Revision>. Convert Dump<T> into Observable<Dump<T>>
+      concatMap(res => (Array.isArray(res) ? from(res) : of(res))),
+      repeatWhen(() => timer(this.pollConfig.cooldown)),
```

Comment:
poll only runs twice and then stops

Reply:
Right, always forget about this behavior of `repeatWhen`, sorry. Fixed.